packages/indexer/src/starknet/parser.ts

```diff
@@ -0,0 +1,50 @@
+import { FieldElement } from "./felt";
+
+interface EventFilterParser {
+  eventFilter(name: string): Filter;
+  // eventParser(name: string): ParsedEvent[];
+}
+
+interface Filter {
+  name: string;
+}
+
+interface Event {
+  // Parsed event properties
+  event: any;
+  transaction: any;
+}
+
+export class Contract implements EventFilterParser {
+  readonly contractAddress: FieldElement;
+  readonly contractAbi: any[];
+
+  constructor(contractAddress: FieldElement, contractAbi: any[]) {
+    this.contractAddress = contractAddress;
+    this.contractAbi = contractAbi;
+  }
+
+  eventFilter(name: string): Filter {
+    // Check if event exists in ABI
+    const event = this.contractAbi.find((item) => item.name === name);
+    if (!event) {
+      throw new Error(`Event ${name} not found in contract ABI`);
+    }
+
+    // Need more clarity on what the DNA filter should look like
+    // Also clarity on the Pedersen hash
+
+    return event;
+  }
+
+  eventParser(name: string): ({ event, transaction }: Event) => Event {
```

Comment:
You can replace `Event` with the real `EventWithTransaction` type, [defined in block.ts](https://github.com/apibara/typescript-sdk/blob/main/packages/indexer/src/starknet/block.ts#L38). The return type in this case will be `any` for now. Later we will make it type safe.

Reply:
For example, parsing the `Transfer` event mentioned above will return an object like `{ from: string, to: string, amount: Uint256 }`, where `Uint256` comes from starknet.js

Reply:
The events will be provided by Apibara. I recommend going through [the getting started tutorial](https://www.apibara.com/docs/getting-started). The parser will be used inside `transform` to decode events from `data: string[]` into typescript objects.

For example, a `Transfer` event with the following data `["0xA", "0xB", "0x1", "0x0"]` will become a typescript object like `{ from: "0xA", to: "0xB", amount: { low: "0x1", high: "0x0" } }`.